src/libraries/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs

```diff
@@ -183,7 +183,10 @@ public static async Task LargeArchive_DataDescriptor_Read_NonZip64_FileLengthGre
                     Assert.Equal(s_bufferSize, read);
                     foreach (byte b in buffer)
                     {
-                        Assert.Equal((byte)'0', b); // the file should be all "0"s 
+                        if (b != '0')
+                        {
+                            Assert.True(false, "The file should be all 0s");
```

Comment:
(nit) Should log the value of `b`. That way if the test fails, you don't need to re-run and debug it to figure out what the value of `b` was.